src/components/ForestFire/ForestFireStateDefinitions.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import Container from '@material-ui/core/Container';
+import Input from '@material-ui/core/Input';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core/styles';
+
+enum attr {
+  COLOR,
+  PROPORTION,
+};
+
+export type ForestFireState = {
+  name: "Tree" | "Burning" | "Ash";
+  color: string;
+  proportion: number;
+};
+
+interface ForestFireStateDefinitionsProps {
+  readonly className?: string;
+  readonly stateDefinitions: ReadonlyArray<ForestFireState>;
+  readonly setStateDefinitions: React.Dispatch<React.SetStateAction<ReadonlyArray<ForestFireState>>>;
+}
+
+const ForestFireStateDefinitions = (props: ForestFireStateDefinitionsProps) => {
+  const { className, stateDefinitions, setStateDefinitions } = props;
+  const classes = useStyles();
+  const onDefinitionChange = (attribute: attr, index: number, e: React.ChangeEvent<HTMLInputElement>) => {
+    setStateDefinitions(definitions => {
+      switch (attribute) {
+        case attr.COLOR:
+          definitions[index].color = e.target.value;
+        case attr.PROPORTION:
+          definitions[index].proportion = Number(e.target.value);
+      };
```

Comment:
There are 3 agent states with  2 editable attributes (color and proportion) per state, stored as an array of objects.
Each edit action needs to specify the target **agent state** and the **attribute**. The **agent state** can be identified using the array index, and I defined `enum attr` to control the *attribute*.
Would like to know if there's a more elegant way to do this - e.g., accessing object attributes using strings?

Reply:
UGH I forgot to add `break` lol

Reply:
also this doesn't really work :(((